src/Tmds.Ssh/UserAuthentication.PublicKeyAuth.cs

```diff
@@ -224,45 +224,127 @@ byte	3
+
+            BigInteger dp = d % (p - BigInteger.One);
+            BigInteger dq = d % (q - BigInteger.One);
+
+            RSAParameters parameters = new()
+            {
+                Modulus = modulus.ToByteArray(isUnsigned: true, isBigEndian: true),
+                Exponent = exponent.ToByteArray(isUnsigned: true, isBigEndian: true),
+                D = d.ToByteArray(isUnsigned: true, isBigEndian: true),
+                InverseQ = inverseQ.ToByteArray(isUnsigned: true, isBigEndian: true),
+                P = p.ToByteArray(isUnsigned: true, isBigEndian: true),
+                Q = q.ToByteArray(isUnsigned: true, isBigEndian: true),
+                DP = dp.ToByteArray(isUnsigned: true, isBigEndian: true),
+                DQ = dq.ToByteArray(isUnsigned: true, isBigEndian: true)
+            };
+            RSA rsa = RSA.Create();
+            try
+            {
+                rsa.ImportParameters(parameters);
+                privateKey = new RsaPrivateKey(rsa);
+                error = null;
+                return true;
+            }
+            catch (Exception ex)
+            {
+                error = new FormatException($"The data can not be parsed into an RSA key.", ex);
+                rsa.Dispose();
+                return false;
+            }
+        }
+
+        private static bool TryParseOpenSshEcdsaKey(Name keyIdentifier, SequenceReader reader, [NotNullWhen(true)] out PrivateKey? privateKey, [NotNullWhen(false)] out Exception? error)
+        {
+            privateKey = null;
+
+            Name curveName = reader.ReadName();
+
+            HashAlgorithmName allowedHashAlgo;
+            ECCurve curve;
+            if (curveName == AlgorithmNames.Nistp256)
+            {
+                allowedHashAlgo = HashAlgorithmName.SHA256;
+                curve = ECCurve.NamedCurves.nistP256;
+            }
+            else if (curveName == AlgorithmNames.Nistp384)
+            {
+                allowedHashAlgo = HashAlgorithmName.SHA384;
+                curve = ECCurve.NamedCurves.nistP384;
+            }
+            else if (curveName == AlgorithmNames.Nistp521)
+            {
+                allowedHashAlgo = HashAlgorithmName.SHA512;
+                curve = ECCurve.NamedCurves.nistP521;
+            }
+            else
+            {
+                error = new FormatException($"ECDSA curve '{curveName}' is unsupported.");
+                return false;
+            }
+
+            ECPoint q = reader.ReadStringAsECPoint();
+            BigInteger d = reader.ReadMPInt();
+
+            ECDsa ecdsa = ECDsa.Create();
+            try
+            {
+                byte[] dBytes = d.ToByteArray(true, true);
+                if (dBytes.Length < q.X!.Length)
                 {
-                    rsa.ImportParameters(parameters);
-                    privateKey = new RsaPrivateKey(rsa);
-                    error = null;
-                    return true;
+                    Array.Resize(ref dBytes, q.X.Length);
```

Comment:
How did you come to add a resize here?

I think if we need to resize, we should be prepending with zeros (for big endian), otherwise we'll change the numeric value.

Reply:
I'm looking, but I don't see where `dBytes` is used.

Reply:
Sorry this was leftover code I was using during testing I forgot to remove.